src/types.rs

```diff
@@ -24,16 +24,29 @@ const fn hex_val(c: u8) -> Option<u8> {
 
 macro_rules! key {
     ($i:ident, $s:expr) => {
-        #[derive(Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        #[repr(transparent)]
+        #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
         /// FuelVM atomic type.
         pub struct $i([u8; $s]);
 
+        impl Default for $i {
+            fn default() -> $i {
+                $i([0u8; $s])
+            }
+        }
+
         key_methods!($i, $s);
 
         #[cfg(feature = "random")]
         impl Distribution<$i> for Standard {
             fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> $i {
-                $i(rng.gen())
+                let mut default: $i = Default::default();
+
+                for i in default.iter_mut() {
+                    *i = rng.gen();
+                }
+
+                default
```

Comment:
Why would we want to generate bytes individually instead of delegating the implementation to `R` so he is aware we are in fact generating `n` bytes?